### PR TITLE
fix: Reduced api hits

### PR DESCRIPTION
**Pull request by Foxify52**
Removed all unnecessary extra hits on the api as detailed in https://github.com/revanced/revanced-manager/issues/545

### CODE REVIEW

**Comment by oSumAtrIX** [comment]
Thanks, has this been tested and confirmed to reduce hits?

**Comment by Foxify52** [comment]
I've fixed one small issue that I've found during testing. Everything is now fully tested and will only hit the API a single time and use the saved response on any consecutive calls.

**Comment by oSumAtrIX** [comment]
Did you check for a difference? Iirc the current code used a cache so it didn't hit the API anyways.

File: lib/services/revanced_api.dart [hunk 1]
```diff
@@ -12,6 +12,12 @@ import 'package:revanced_manager/models/patch.dart';
 import 'package:revanced_manager/utils/check_for_gms.dart';
 import 'package:timeago/timeago.dart';
 
+Map<String, List<dynamic>> contributors = {};
```
Comment (oSumAtrIX) [inline lib/services/revanced_api.dart:15]: I am not sure if it is traditional to abuse a map in this file for caching purposes. Instead a dedicated middleware should cache the REST api calls and IIrc there is already one. The question would be why it doesn't work.

**Comment by Sculas** [comment]
@alexandreteles Does the API support ETags? If so, you could use that for caching the requests via an ETag-compatible DIO middleware. Of course, this requires the server to have a cache layer for the entities it's returning (it **should** already have that!). The good thing about this is that if the data ever becomes stale, it will automatically be updated. Otherwise, a `304 Not Modified` is returned with no body, which saves on bandwidth as well.

This PR works as well, though not the cleanest option in my opinion.

**Comment by oSumAtrIX** [comment]
@Sculas I agree, any ideas on https://github.com/revanced/revanced-manager/pull/774#discussion_r1161267841

**Comment by Sculas** [comment]
The server already returns ETags, so literally, all that needs to be done is that this library needs to be added: [dio_cache_interceptor](https://pub.dev/packages/dio_cache_interceptor).

**Comment by oSumAtrIX** [comment]
@Sculas do you happen to know why the current cache fails to work?

**Comment by Sculas** [comment]
I'm not sure why the existing cache impl doesn't work. But using the one I sent should work, though.

**Comment by Aunali321** [comment]
Because it does not use dio_cache_interceptor as @Sculas said. It's using another library. But it's been replaced in this commit https://github.com/revanced/revanced-manager/commit/c2021d508eb05bfd48d3b65d8ff440dd41311273 of the `improve-ux` branch which is not yet in production.

**Comment by oSumAtrIX** [comment]
Another library or not, it's a cache, but it does not work as expected leading to my question

**Comment by Aunali321** [comment]
The older library did not support Etags.

**Comment by oSumAtrIX** [comment]
And how was it intending to cache?

**Comment by Aunali321** [comment]
> And how was it intending to cache?

https://pub.dev/packages/dio_http_cache

**Comment by oSumAtrIX** [comment]
I don't follow, there's a cache implementation but it does not cache, which is my question

**Comment by Foxify52** [comment]
Perhaps the reason it doesn't cache is because it was never set up right to begin with.

**Comment by oSumAtrIX** [comment]
In that case, this PR would be working around that issue. Instead, it should either replace the incomplete implementation, or fix it.

**Pull request closed without merging**